### ISSUE TITLE
Helm 2.16 - context deadline exceeded with helm.NewClient in-cluster

### ISSUE THREAD

**Issue by mmailhos** [helm/helm#7434]
When running a simple `helm.NewClient` with Helm 2.16 in a Kubernetes cluster, I am unable to even `Ping` the tiller server. The err is `context deadline exceeded`.

## Versions

Output of `helm version`: v2.16.1
Output of `kubectl version`: 1.17.0
Cloud Provider/Platform (AKS, GKE, Minikube etc.):  minikube v1.6.2

## Test code

````
package main

import (
	"fmt"

	"k8s.io/helm/pkg/helm"
)

func main() {
	c := helm.NewClient(helm.Host("tiller-deploy"))
	if err := c.PingTiller(); err != nil {
		panic(err) // Panic here when running in cluster
	}
         // OK when using localhost:44134 after a port-forward when running outside of the cluster
	fmt.Println("ok!") 
}
````

## How to reproduce

You can find the Dockerfile and deployment.yaml in this gist [here](https://gist.github.com/mmailhos/f22f03e12653a16648cdc94a68976dfa)

```
$ minikube start
$ kubectl create clusterrolebinding add-on-cluster-admin --clusterrole=cluster-admin --serviceaccount=default:default
$ helm init --tiller-namespace default
$ helm version --tiller-namespace default
Client: &version.Version{SemVer:"v2.16.1", GitCommit:"bbdfe5e7803a12bbdf97e94cd847859890cf4050", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.16.1", GitCommit:"bbdfe5e7803a12bbdf97e94cd847859890cf4050", GitTreeState:"clean"}
```

Build with Go 1.13.4 for Helm 2.16.1 with `dep`
```
$ dep init
$ dep ensure -add k8s.io/helm/pkg/helm
$ eval $(minikube docker-env)
$ GOOS=linux docker build -t test .
$ minikube kubectl apply -f deployment.yaml
```

```
kubectl logs test-deployment-xxx
panic: context deadline exceeded
```

The tiller server is exposed on 44134 with a proper Kubernetes services.
Doing a port-forward and running the code locally works.
The tiller server is accessible with telnet 44134 from a ubuntu pod running in the same namespace. 
This looks to be the same issue as https://github.com/helm/helm/issues/5515

**Comment by bacongobbler** [comment]
Did you make sure to set a gRPC connection timeout? If I'm not mistaken, gRPC will disconnect immediately if this parameter is not set.

https://github.com/helm/helm/blob/63e1c9bc7f6a88efe40f5badcc0766e6748283c9/cmd/helm/helm.go#L371

**Comment by mmailhos** [comment]
> Did you make sure to set a gRPC connection timeout? If I'm not mistaken, gRPC will disconnect immediately if this parameter is not set.
> 
> https://github.com/helm/helm/blob/63e1c9bc7f6a88efe40f5badcc0766e6748283c9/cmd/helm/helm.go#L371

Yes, it just get the same error after `n` seconds. Do I need to have a TLS config though?
Weird also that the same code behave differently between inside and outside Kubernetes also.
